File: KIJUNG/PRO_WEEK_04/소수찾기.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <math.h>

using namespace std;

bool isPrime(int n){
    if(n<=1) return false;
    for(int i=2;i<=sqrt(n);i++){
        if(n%i == 0) return false;
    }
    return true;
}

int solution(string numbers) {
    vector<int> nums;
    int answer = 0;

    sort(numbers.begin(),numbers.end());
    do{
        for(int i=1;i<=numbers.size();i++){
            nums.push_back(stoi(numbers.substr(0,i)));
        }
    }while(next_permutation(numbers.begin(), numbers.end()));
    sort(nums.begin(), nums.end());
    nums.erase(unique(nums.begin(), nums.end()), nums.end());

    for(int num: nums){
        if(isPrime(num)) answer++;
    }
    
    return answer;
}

File: DONGUK/프로그래머스_WEEK_15/소수 찾기.py
def solution(n):
    answer = 0
    
    dp = [1] * (n+1)
    
    for i in range(2, int(n ** (1/2)) + 1):
        if dp[i] == 1:
            j = 2
            while i * j <= n:
                dp[i*j] = 0
                j += 1
            
    for i in range(2, n+1):
        if dp[i] == 1:
            answer += 1
            
    return answer